src/Utils/StyleAttributesUtils.php

```diff
@@ -204,32 +223,50 @@ public static function get_background_color_class_and_style( $attributes ) {
 	/**
 	 * Get class and style for border-color from attributes.
 	 *
+	 * Data passed to this function is not always consistent. It can be:
+	 * Linked - preset color: $attributes['borderColor'] => 'luminous-vivid-orange'.
+	 * Linked - custom color: $attributes['style']['border']['color'] => '#681228'.
+	 * Unlinked - preset color: $attributes['style']['border']['top']['color'] => 'var:preset|color|luminous-vivid-orange'
+	 * Unlinked - custom color: $attributes['style']['border']['top']['color'] => '#681228'.
+	 *
 	 * @param array $attributes Block attributes.
 	 *
 	 * @return (array | null)
 	 */
 	public static function get_border_color_class_and_style( $attributes ) {
 
-		$border_color = $attributes['borderColor'] ?? '';
+		$border_color_linked_preset = $attributes['borderColor'] ?? '';
+		$border_color_linked_custom = $attributes['style']['border']['color'] ?? '';
+		$custom_border              = $attributes['style']['border'] ?? '';
 
-		$custom_border_color = $attributes['style']['border']['color'] ?? '';
+		$border_color_class = '';
+		$border_color_css   = '';
 
-		if ( ! $border_color && '' === $custom_border_color ) {
-			return null;
+		if ( $border_color_linked_preset ) {
+			// Linked preset color.
+			$border_color_class = sprintf( 'has-border-color has-%s-border-color', $border_color_linked_preset );
+		} elseif ( $border_color_linked_custom ) {
+			// Linked custom color.
+			$border_color_css .= 'border-color:' . $border_color_linked_custom . ';';
+		} else {
+			// Unlinked.
+			if ( '' !== $custom_border ) {
```

Comment:
Instead of checking that is not an empty string, I wonder if it would be safer to check that `$custom_border` is an array. What do you think?

Reply:
That does make more sense. Thank you for this feedback :)